{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# indexPdb.py\n",
    "# ----------------------------------------------------------------------\n",
    "# Copyright (2010) Aram Davtyan and Garegin Papoian\n",
    "#\n",
    "# Papoian's Group, University of Maryland at Collage Park\n",
    "# http://papoian.chem.umd.edu/\n",
    "#\n",
    "# Last Update: 07/18/2011\n",
    "# -------------------------------------------------------------------------\n",
    "\n",
    "#import sys\n",
    "\n",
    "#if len(sys.argv)!=5:\n",
    "#\tprint\n",
    "#\tprint \" > \" + sys.argv[0] + \" fasta.file pdb_file index_file chain_id\"\n",
    "#\tprint\n",
    "#\texit()\n",
    "\n",
    "from Bio import pairwise2\n",
    "from Bio.PDB.PDBParser import PDBParser\n",
    "from Bio import SeqIO\n",
    "\n",
    "def three2one(prot):\n",
    "    \"\"\" translate a protein sequence from 3 to 1 letter code\"\"\"\n",
    "\n",
    "    code = {\"GLY\" : \"G\", \"ALA\" : \"A\", \"LEU\" : \"L\", \"ILE\" : \"I\",\n",
    "            \"ARG\" : \"R\", \"LYS\" : \"K\", \"MET\" : \"M\", \"CYS\" : \"C\",\n",
    "            \"TYR\" : \"Y\", \"THR\" : \"T\", \"PRO\" : \"P\", \"SER\" : \"S\",\n",
    "            \"TRP\" : \"W\", \"ASP\" : \"D\", \"GLU\" : \"E\", \"ASN\" : \"N\",\n",
    "            \"GLN\" : \"Q\", \"PHE\" : \"F\", \"HIS\" : \"H\", \"VAL\" : \"V\",\n",
    "            \"M3L\" : \"K\", \"MSE\" : \"M\", \"CAS\" : \"C\" }\n",
    "\n",
    "    newprot = \"\"\n",
    "    for a in prot:\n",
    "        newprot += code.get(a, \"X\")\n",
    "\n",
    "    return newprot\n",
    "\n",
    "def getListOfValidAlignments(alignments, pdb_indexes):\n",
    "    ialigns = []\n",
    "    for i in range(0, len(alignments)):\n",
    "        alignment = alignments[i]\n",
    "        alseq_fasta = alignment[0]\n",
    "        alseq_pdb = alignment[1]\n",
    "\n",
    "        if len(alseq_fasta)!=len(alseq_pdb):\n",
    "            print(\"Error using alignment too\")\n",
    "            print()\n",
    "            exit()\n",
    "\n",
    "        last_i_pdb = -1\n",
    "        last_j = -1\n",
    "        i_pdb = 0\n",
    "        valid = True\n",
    "        for j in range(0, len(alseq_pdb)):\n",
    "                if alseq_pdb[j]!='-':\n",
    "    #                                if last_i_pdb!=-1 and j - last_j > i_pdb - last_i_pdb:\n",
    "                        if last_i_pdb!=-1 and j - last_j != pdb_indexes[i_pdb] - pdb_indexes[last_i_pdb]:\n",
    "                                valid = False\n",
    "                                break\n",
    "                        last_i_pdb = i_pdb\n",
    "                        last_j = j\n",
    "                        i_pdb = i_pdb + 1\n",
    "        if valid:\n",
    "                ialigns.append(i)\n",
    "    return ialigns\n",
    "\n",
    "def getFastaSequance(fasta_file):\n",
    "    inFASTA=open(fasta_file, 'r')\n",
    "    inseq=SeqIO.read(inFASTA,'fasta')\n",
    "\n",
    "    return str(inseq.seq)\n",
    "\n",
    "def getPdbSequance(pdb_file, chain_id):\n",
    "    pdb_indexes = []\n",
    "    pdb_sequance = []\n",
    "\n",
    "    p = PDBParser(PERMISSIVE=1)\n",
    "    s = p.get_structure(\"\",  pdb_file)\n",
    "    pdb_id = pdb_file[0:-4]\n",
    "\n",
    "    if not s[0].has_id(chain_id):\n",
    "        print(\"PDB \"+pdb_id+\" doesn't have chain with id \"+chain_id)\n",
    "        print()\n",
    "        exit()\n",
    "\n",
    "    chain = s[0][chain_id]\n",
    "\n",
    "    ires = 0\n",
    "    for res in chain:\n",
    "        is_regular_res = res.has_id('N') and res.has_id('CA') and res.has_id('C') and (res.get_resname()=='GLY' or res.has_id('CB'))\n",
    "        res_id = res.get_id()[0]\n",
    "        if (res_id ==' ' or res_id =='H_MSE' or res_id =='H_M3L' or res_id =='H_CAS') and is_regular_res:\n",
    "                ires = ires + 1\n",
    "                res_name = res.get_resname()\n",
    "                residue_no = res.get_id()[1]\n",
    "                pdb_sequance.append(res_name)\n",
    "                pdb_indexes.append(residue_no)\n",
    "        elif res_id !='W':\n",
    "                print(\"Unknown residue in \"+pdb_id+\" with res_id \"+res_id)\n",
    "\n",
    "    pdb_seq = three2one(pdb_sequance)\n",
    "\n",
    "    return pdb_seq, pdb_indexes\n",
    "\n",
    "def getIndexArray(alignment, pdb_indexes):\n",
    "    alseq_fasta = alignment[0]\n",
    "    alseq_pdb = alignment[1]\n",
    "\n",
    "    index_array = []\n",
    "    i_fasta = 0\n",
    "    i_pdb = 0\n",
    "    for i in range(0, len(alseq_fasta)):\n",
    "        if alseq_fasta[i]!='-':\n",
    "            index_pdb = -1\n",
    "            if alseq_pdb[i]!='-': index_pdb = pdb_indexes[i_pdb]\n",
    "            index_array.append([i_fasta, index_pdb, alseq_fasta[i]])\n",
    "            i_fasta = i_fasta + 1\n",
    "        if alseq_pdb[i]!='-':\n",
    "            i_pdb = i_pdb + 1\n",
    "\n",
    "    return index_array\n",
    "\n",
    "def writeIndexFile(fasta_file, pdb_file, index_file, chain_id):\n",
    "#\tfrom Bio import pairwise2\n",
    "#\tfrom Bio.PDB.PDBParser import PDBParser\n",
    "#\tfrom Bio import SeqIO\n",
    "\n",
    "    pdb_id = pdb_file[0:-4]\n",
    "\n",
    "#fasta_file = sys.argv[1]\n",
    "#pdb_id = sys.argv[2]\n",
    "#pdb_file = pdb_id+\".pdb\"\n",
    "#index_file = sys.argv[3]\n",
    "#chain_id = sys.argv[4]\n",
    "\n",
    "    fasta_seq = \"\"\n",
    "    pdb_seq = \"\"\n",
    "    pdb_indexes = []\n",
    "    answer = \"\"\n",
    "    shift = 0\n",
    "    index_list = []\n",
    "\n",
    "    fasta_seq = getFastaSequance(fasta_file)\n",
    "    pdb_seq, pdb_indexes = getPdbSequance(pdb_file, chain_id)\n",
    "\n",
    "    print()\n",
    "    print(fasta_seq)\n",
    "    print(len(fasta_seq))\n",
    "    print(pdb_seq)\n",
    "    print(pdb_indexes)\n",
    "    print(len(pdb_seq))\n",
    "\n",
    "    print()\n",
    "    print()\n",
    "\n",
    "    if len(fasta_seq)==len(pdb_seq) and pdb_indexes[0]==1 and pdb_indexes[-1]==len(pdb_seq):\n",
    "        print(\"FULLMATCH\")\n",
    "        answer = \"FULLMATCH\"\n",
    "    elif len(fasta_seq)==len(pdb_seq) and pdb_indexes[-1]-pdb_indexes[0]+1==len(pdb_seq):\n",
    "        print(\"Indexes are simply shifted by \" + str(pdb_indexes[0]-1))\n",
    "        answer = \"SHIFT\"\n",
    "        shift = pdb_indexes[0]-1\n",
    "    elif len(fasta_seq)==len(pdb_seq) and fasta_seq==pdb_seq:\n",
    "        print(\"Number is messed up\")\n",
    "        print(\"Same length\")\n",
    "        answer = \"INDEXED\"\n",
    "        for i in range(0, len(fasta_seq)):\n",
    "            if i!=0 and pdb_indexes[i]<= pdb_indexes[i-1]:\n",
    "                answer = \"SKIP\"\n",
    "                index_list = []\n",
    "                break\n",
    "            index_list.append([ i, pdb_indexes[i], fasta_seq[i] ])\n",
    "    else:\n",
    "        alignments = pairwise2.align.globalms(fasta_seq, pdb_seq, 2, -1, -0.5, -0.1)\n",
    "        #print alignments\n",
    "        #print len(alignments)\n",
    "        #print\n",
    "\n",
    "        alist = getListOfValidAlignments(alignments, pdb_indexes)\n",
    "        #print len(alist), alist\n",
    "\n",
    "        if len(alist)==1:\n",
    "            answer = \"INDEXED\"\n",
    "            index_list = getIndexArray(alignments[alist[0]], pdb_indexes)\n",
    "        elif len(alist)>1:\n",
    "            answer = \"SKIP\"\n",
    "        elif len(alist)==0:\n",
    "            answer = \"SKIP\"\n",
    "            #alignments = pairwise2.align.globalxx(fasta_seq, pdb_seq)\n",
    "            #print\n",
    "            #print alignments\n",
    "                #print len(alignments)\n",
    "            #print\n",
    "\n",
    "            #alist2 = getListOfValidAlignments(alignments, pdb_indexes)\n",
    "                #print len(alist2), alist2\n",
    "\n",
    "            #if len(alist2)==1:\n",
    "            #\tanswer = \"INDEXED\"\n",
    "            #\tindex_list = getIndexArray(alignments[alist2[0]], pdb_indexes)\n",
    "            #elif len(alist2)>1:\n",
    "            #\tanswer = \"SKIP\"\n",
    "            #elif len(alist2)==0:\n",
    "            #\tanswer = \"SKIP\"\n",
    "\n",
    "    out = open(index_file, 'w')\n",
    "    out.write(answer)\n",
    "    if answer==\"SHIFT\":\n",
    "        out.write(\"\\n\")\n",
    "        out.write(str(shift))\n",
    "    elif answer==\"INDEXED\":\n",
    "        for ind in index_list:\n",
    "            out.write(\"\\n\")\n",
    "            out.write(str(ind[0]+1))\n",
    "            out.write(\" \")\n",
    "            out.write(str(ind[1]))\n",
    "            out.write(\" \")\n",
    "            out.write(ind[2])\n",
    "    out.close()\n",
    "\n",
    "\n",
    "#fasta_file = sys.argv[1]\n",
    "#pdb_id = sys.argv[2]\n",
    "#pdb_file = pdb_id+\".pdb\"\n",
    "#index_file = sys.argv[3]\n",
    "#chain_id = sys.argv[4]\n",
    "\n",
    "#writeIndexFile(fasta_file, pdb_file, index_file, chain_id)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/weilu/Research/server/jun_2019/simluation_hybrid/frag_database/6e67A_HA/globularPart'"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.path.dirname(fasta_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unknown residue in /Users/weilu/openmmawsem/PDBs/5CXV with res_id H_0HK\n",
      "Unknown residue in /Users/weilu/openmmawsem/PDBs/5CXV with res_id H_Y01\n",
      "Unknown residue in /Users/weilu/openmmawsem/PDBs/5CXV with res_id H_EDO\n",
      "Unknown residue in /Users/weilu/openmmawsem/PDBs/5CXV with res_id H_PGE\n",
      "Unknown residue in /Users/weilu/openmmawsem/PDBs/5CXV with res_id H_EDO\n",
      "Unknown residue in /Users/weilu/openmmawsem/PDBs/5CXV with res_id H_GOL\n",
      "Unknown residue in /Users/weilu/openmmawsem/PDBs/5CXV with res_id H_EDO\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/weilu/anaconda3/envs/py36/lib/python3.6/site-packages/Bio/PDB/StructureBuilder.py:91: PDBConstructionWarning: WARNING: Chain A is discontinuous at line 4123.\n",
      "  PDBConstructionWarning)\n",
      "/Users/weilu/anaconda3/envs/py36/lib/python3.6/site-packages/Bio/PDB/StructureBuilder.py:91: PDBConstructionWarning: WARNING: Chain C is discontinuous at line 4218.\n",
      "  PDBConstructionWarning)\n"
     ]
    }
   ],
   "source": [
    "fasta_file = \"/Users/weilu/Research/server/jun_2019/simluation_hybrid/frag_database/6e67A_HA/globularPart/5cxva.fasta\"\n",
    "pdb_file = \"/Users/weilu/openmmawsem/PDBs/5CXV.pdb\"\n",
    "# pdb_file = \"/Users/weilu/Research/server/jun_2019/simluation_hybrid/frag_database/6e67A_HA/globularPart/5cxv.pdb\"\n",
    "index_file = \"/Users/weilu/openmmawsem/Indices/5cxva.index\"\n",
    "chain_id = \"A\"\n",
    "\n",
    "fasta_seq = \"\"\n",
    "pdb_seq = \"\"\n",
    "pdb_indexes = []\n",
    "answer = \"\"\n",
    "shift = 0\n",
    "index_list = []\n",
    "\n",
    "fasta_seq = getFastaSequance(fasta_file)\n",
    "pdb_seq, pdb_indexes = getPdbSequance(pdb_file, chain_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1160 355 need reorder\n"
     ]
    }
   ],
   "source": [
    "pre = pdb_indexes[0]\n",
    "for i in pdb_indexes[1:]:\n",
    "    if i < pre:\n",
    "        print(pre, i, \"need reorder\")\n",
    "        break\n",
    "    pre = i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[20,\n",
       " 21,\n",
       " 22,\n",
       " 23,\n",
       " 24,\n",
       " 25,\n",
       " 26,\n",
       " 27,\n",
       " 28,\n",
       " 29,\n",
       " 30,\n",
       " 31,\n",
       " 32,\n",
       " 33,\n",
       " 34,\n",
       " 35,\n",
       " 36,\n",
       " 37,\n",
       " 38,\n",
       " 39,\n",
       " 40,\n",
       " 41,\n",
       " 42,\n",
       " 43,\n",
       " 44,\n",
       " 45,\n",
       " 46,\n",
       " 47,\n",
       " 48,\n",
       " 49,\n",
       " 50,\n",
       " 51,\n",
       " 52,\n",
       " 53,\n",
       " 54,\n",
       " 55,\n",
       " 56,\n",
       " 57,\n",
       " 58,\n",
       " 59,\n",
       " 60,\n",
       " 61,\n",
       " 62,\n",
       " 63,\n",
       " 64,\n",
       " 65,\n",
       " 66,\n",
       " 67,\n",
       " 68,\n",
       " 69,\n",
       " 70,\n",
       " 71,\n",
       " 72,\n",
       " 73,\n",
       " 74,\n",
       " 75,\n",
       " 76,\n",
       " 77,\n",
       " 78,\n",
       " 79,\n",
       " 80,\n",
       " 81,\n",
       " 82,\n",
       " 83,\n",
       " 84,\n",
       " 85,\n",
       " 86,\n",
       " 87,\n",
       " 88,\n",
       " 89,\n",
       " 90,\n",
       " 91,\n",
       " 92,\n",
       " 93,\n",
       " 94,\n",
       " 95,\n",
       " 96,\n",
       " 97,\n",
       " 98,\n",
       " 99,\n",
       " 100,\n",
       " 101,\n",
       " 102,\n",
       " 103,\n",
       " 104,\n",
       " 105,\n",
       " 106,\n",
       " 107,\n",
       " 108,\n",
       " 109,\n",
       " 110,\n",
       " 111,\n",
       " 112,\n",
       " 113,\n",
       " 114,\n",
       " 115,\n",
       " 116,\n",
       " 117,\n",
       " 118,\n",
       " 119,\n",
       " 120,\n",
       " 121,\n",
       " 122,\n",
       " 123,\n",
       " 124,\n",
       " 125,\n",
       " 126,\n",
       " 127,\n",
       " 128,\n",
       " 129,\n",
       " 130,\n",
       " 131,\n",
       " 132,\n",
       " 133,\n",
       " 134,\n",
       " 135,\n",
       " 136,\n",
       " 137,\n",
       " 138,\n",
       " 139,\n",
       " 140,\n",
       " 141,\n",
       " 142,\n",
       " 143,\n",
       " 144,\n",
       " 145,\n",
       " 146,\n",
       " 147,\n",
       " 148,\n",
       " 149,\n",
       " 150,\n",
       " 151,\n",
       " 152,\n",
       " 153,\n",
       " 154,\n",
       " 155,\n",
       " 156,\n",
       " 157,\n",
       " 158,\n",
       " 159,\n",
       " 160,\n",
       " 161,\n",
       " 162,\n",
       " 163,\n",
       " 164,\n",
       " 165,\n",
       " 166,\n",
       " 167,\n",
       " 168,\n",
       " 169,\n",
       " 170,\n",
       " 171,\n",
       " 172,\n",
       " 173,\n",
       " 174,\n",
       " 175,\n",
       " 176,\n",
       " 177,\n",
       " 178,\n",
       " 179,\n",
       " 180,\n",
       " 181,\n",
       " 182,\n",
       " 183,\n",
       " 184,\n",
       " 185,\n",
       " 186,\n",
       " 187,\n",
       " 188,\n",
       " 189,\n",
       " 190,\n",
       " 191,\n",
       " 192,\n",
       " 193,\n",
       " 194,\n",
       " 195,\n",
       " 196,\n",
       " 197,\n",
       " 198,\n",
       " 199,\n",
       " 200,\n",
       " 201,\n",
       " 202,\n",
       " 203,\n",
       " 204,\n",
       " 205,\n",
       " 206,\n",
       " 207,\n",
       " 208,\n",
       " 209,\n",
       " 210,\n",
       " 211,\n",
       " 212,\n",
       " 213,\n",
       " 214,\n",
       " 215,\n",
       " 216,\n",
       " 217,\n",
       " 218,\n",
       " 1001,\n",
       " 1002,\n",
       " 1003,\n",
       " 1004,\n",
       " 1005,\n",
       " 1006,\n",
       " 1007,\n",
       " 1008,\n",
       " 1009,\n",
       " 1010,\n",
       " 1011,\n",
       " 1012,\n",
       " 1013,\n",
       " 1014,\n",
       " 1015,\n",
       " 1016,\n",
       " 1017,\n",
       " 1018,\n",
       " 1019,\n",
       " 1020,\n",
       " 1021,\n",
       " 1022,\n",
       " 1023,\n",
       " 1024,\n",
       " 1025,\n",
       " 1026,\n",
       " 1027,\n",
       " 1028,\n",
       " 1029,\n",
       " 1030,\n",
       " 1031,\n",
       " 1032,\n",
       " 1033,\n",
       " 1034,\n",
       " 1035,\n",
       " 1036,\n",
       " 1037,\n",
       " 1038,\n",
       " 1039,\n",
       " 1040,\n",
       " 1041,\n",
       " 1042,\n",
       " 1043,\n",
       " 1044,\n",
       " 1045,\n",
       " 1046,\n",
       " 1047,\n",
       " 1048,\n",
       " 1049,\n",
       " 1050,\n",
       " 1051,\n",
       " 1052,\n",
       " 1053,\n",
       " 1054,\n",
       " 1055,\n",
       " 1056,\n",
       " 1057,\n",
       " 1058,\n",
       " 1059,\n",
       " 1060,\n",
       " 1061,\n",
       " 1062,\n",
       " 1063,\n",
       " 1064,\n",
       " 1065,\n",
       " 1066,\n",
       " 1067,\n",
       " 1068,\n",
       " 1069,\n",
       " 1070,\n",
       " 1071,\n",
       " 1072,\n",
       " 1073,\n",
       " 1074,\n",
       " 1075,\n",
       " 1076,\n",
       " 1077,\n",
       " 1078,\n",
       " 1079,\n",
       " 1080,\n",
       " 1081,\n",
       " 1082,\n",
       " 1083,\n",
       " 1084,\n",
       " 1085,\n",
       " 1086,\n",
       " 1087,\n",
       " 1088,\n",
       " 1089,\n",
       " 1090,\n",
       " 1091,\n",
       " 1092,\n",
       " 1093,\n",
       " 1094,\n",
       " 1095,\n",
       " 1096,\n",
       " 1097,\n",
       " 1098,\n",
       " 1099,\n",
       " 1100,\n",
       " 1101,\n",
       " 1102,\n",
       " 1103,\n",
       " 1104,\n",
       " 1105,\n",
       " 1106,\n",
       " 1107,\n",
       " 1108,\n",
       " 1109,\n",
       " 1110,\n",
       " 1111,\n",
       " 1112,\n",
       " 1113,\n",
       " 1114,\n",
       " 1115,\n",
       " 1116,\n",
       " 1117,\n",
       " 1118,\n",
       " 1119,\n",
       " 1120,\n",
       " 1121,\n",
       " 1122,\n",
       " 1123,\n",
       " 1124,\n",
       " 1125,\n",
       " 1126,\n",
       " 1127,\n",
       " 1128,\n",
       " 1129,\n",
       " 1130,\n",
       " 1131,\n",
       " 1132,\n",
       " 1133,\n",
       " 1134,\n",
       " 1135,\n",
       " 1136,\n",
       " 1137,\n",
       " 1138,\n",
       " 1139,\n",
       " 1140,\n",
       " 1141,\n",
       " 1142,\n",
       " 1143,\n",
       " 1144,\n",
       " 1145,\n",
       " 1146,\n",
       " 1147,\n",
       " 1148,\n",
       " 1149,\n",
       " 1150,\n",
       " 1151,\n",
       " 1152,\n",
       " 1153,\n",
       " 1154,\n",
       " 1155,\n",
       " 1156,\n",
       " 1157,\n",
       " 1158,\n",
       " 1159,\n",
       " 1160,\n",
       " 355,\n",
       " 356,\n",
       " 357,\n",
       " 358,\n",
       " 359,\n",
       " 360,\n",
       " 361,\n",
       " 362,\n",
       " 363,\n",
       " 364,\n",
       " 365,\n",
       " 366,\n",
       " 367,\n",
       " 368,\n",
       " 369,\n",
       " 370,\n",
       " 371,\n",
       " 372,\n",
       " 373,\n",
       " 374,\n",
       " 375,\n",
       " 376,\n",
       " 377,\n",
       " 378,\n",
       " 379,\n",
       " 380,\n",
       " 381,\n",
       " 382,\n",
       " 383,\n",
       " 384,\n",
       " 385,\n",
       " 386,\n",
       " 387,\n",
       " 388,\n",
       " 389,\n",
       " 390,\n",
       " 391,\n",
       " 392,\n",
       " 393,\n",
       " 394,\n",
       " 395,\n",
       " 396,\n",
       " 397,\n",
       " 398,\n",
       " 399,\n",
       " 400,\n",
       " 401,\n",
       " 402,\n",
       " 403,\n",
       " 404,\n",
       " 405,\n",
       " 406,\n",
       " 407,\n",
       " 408,\n",
       " 409,\n",
       " 410,\n",
       " 411,\n",
       " 412,\n",
       " 413,\n",
       " 414,\n",
       " 415,\n",
       " 416,\n",
       " 417,\n",
       " 418,\n",
       " 419,\n",
       " 420,\n",
       " 421,\n",
       " 422,\n",
       " 423,\n",
       " 424,\n",
       " 425,\n",
       " 426,\n",
       " 427,\n",
       " 428,\n",
       " 429,\n",
       " 430,\n",
       " 431,\n",
       " 432,\n",
       " 433,\n",
       " 434,\n",
       " 435,\n",
       " 436,\n",
       " 437,\n",
       " 438,\n",
       " 439]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pdb_indexes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "MKTIIALSYIFCLVFADYKDDDDAAAQTSAPPAVSPQITVLAPGKGPWQVAFIGITTGLLSLATVTGNLLVLISFKVNTELKTVNNYFLLSLACADLIIGTFSMNLYTTYLLMGHWALGTLACDLWLALDYVASQASVMNLLLISFDRYFSVTRPLSYRAKRTPRRAALMIGLAWLVSFVLWAPAILFWQYLVGERTVLAGQCYIQFLSQPIITFGTAMAAFYLPVTVMCTLYWRIYRETENRNIFEMLRIDEGLRLKIYKDTEGYYTIGIGHLLTKSPSLNAAKSELDKAIGRNTNGVITKDEAEKLFNQDVDAAVRGILRNAKLKPVYDSLDAVRRAALINMVFQMGETGVAGFTNSLRMLQQKRWDEAAVNLAKSRWYNQTPNRAKRVITTFRTGTWDAYFSLVKEKKAARTLSAILLAFILTWTPYNIMVLVSTFCKDCVPETLWELGYWLCYVNSTINPMCYALCNKAFRDTFRLLLLCRWDKRRWRKIPKRPGSVHRTPSRQCHHHHHH\n",
      "515\n",
      "KGPWQVAFIGITTGLLSLATVTGNLLVLISFKVNTELKTVNNYFLLSLACADLIIGTFSMNLYTTYLLMGHWALGTLACDLWLALDYVASQASVMNLLLISFDRYFSVTRPLSYRAKRTPRRAALMIGLAWLVSFVLWAPAILFWQYLVGERTVLAGQCYIQFLSQPIITFGTAMAAFYLPVTVMCTLYWRIYRETENRNIFEMLRIDEGLRLKIYKDTEGYYTIGIGHLLTKSPSLNAAKSELDKAIGRNTNGVITKDEAEKLFNQDVDAAVRGILRNAKLKPVYDSLDAVRRAALINMVFQMGETGVAGFTNSLRMLQQKRWDEAAVNLAKSRWYNQTPNRAKRVITTFRTGTWDAYFSLVKEKKAARTLSAILLAFILTWTPYNIMVLVSTFCKDCVPETLWELGYWLCYVNSTINPMCYALCNKAFRDTFRLLLLCRWDK\n",
      "[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215, 216, 217, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 252, 253, 254, 255, 256, 257, 258, 259, 260, 261, 262, 263, 264, 265, 266, 267, 268, 269, 270, 271, 272, 273, 274, 275, 276, 277, 278, 279, 280, 281, 282, 283, 284, 285, 286, 287, 288, 289, 290, 291, 292, 293, 294, 295, 296, 297, 298, 299, 300, 301, 302, 303, 304, 305, 306, 307, 308, 309, 310, 311, 312, 313, 314, 315, 316, 317, 318, 319, 320, 321, 322, 323, 324, 325, 326, 327, 328, 329, 330, 331, 332, 333, 334, 335, 336, 337, 338, 339, 340, 341, 342, 343, 344, 345, 346, 347, 348, 349, 350, 351, 352, 353, 354, 355, 356, 357, 358, 359, 360, 361, 362, 363, 364, 365, 366, 367, 368, 369, 370, 371, 372, 373, 374, 375, 376, 377, 378, 379, 380, 381, 382, 383, 384, 385, 386, 387, 388, 389, 390, 391, 392, 393, 394, 395, 396, 397, 398, 399, 400, 401, 402, 403, 404, 405, 406, 407, 408, 409, 410, 411, 412, 413, 414, 415, 416, 417, 418, 419, 420, 421, 422, 423, 424, 425, 426, 427, 428, 429, 430, 431, 432, 433, 434, 435, 436, 437, 438, 439, 440, 441, 442, 443, 444]\n",
      "444\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "print()\n",
    "print(fasta_seq)\n",
    "print(len(fasta_seq))\n",
    "print(pdb_seq)\n",
    "print(pdb_indexes)\n",
    "print(len(pdb_seq))\n",
    "\n",
    "print()\n",
    "print()\n",
    "\n",
    "if len(fasta_seq)==len(pdb_seq) and pdb_indexes[0]==1 and pdb_indexes[-1]==len(pdb_seq):\n",
    "    print(\"FULLMATCH\")\n",
    "    answer = \"FULLMATCH\"\n",
    "elif len(fasta_seq)==len(pdb_seq) and pdb_indexes[-1]-pdb_indexes[0]+1==len(pdb_seq):\n",
    "    print(\"Indexes are simply shifted by \" + str(pdb_indexes[0]-1))\n",
    "    answer = \"SHIFT\"\n",
    "    shift = pdb_indexes[0]-1\n",
    "elif len(fasta_seq)==len(pdb_seq) and fasta_seq==pdb_seq:\n",
    "    print(\"Number is messed up\")\n",
    "    print(\"Same length\")\n",
    "    answer = \"INDEXED\"\n",
    "    for i in range(0, len(fasta_seq)):\n",
    "        if i!=0 and pdb_indexes[i]<= pdb_indexes[i-1]:\n",
    "            answer = \"SKIP\"\n",
    "            index_list = []\n",
    "            break\n",
    "        index_list.append([ i, pdb_indexes[i], fasta_seq[i] ])\n",
    "else:\n",
    "    alignments = pairwise2.align.globalms(fasta_seq, pdb_seq, 2, -1, -0.5, -0.1)\n",
    "    #print alignments\n",
    "    #print len(alignments)\n",
    "    #print\n",
    "\n",
    "    alist = getListOfValidAlignments(alignments, pdb_indexes)\n",
    "    #print len(alist), alist\n",
    "\n",
    "    if len(alist)==1:\n",
    "        answer = \"INDEXED\"\n",
    "        index_list = getIndexArray(alignments[alist[0]], pdb_indexes)\n",
    "    elif len(alist)>1:\n",
    "        answer = \"SKIP\"\n",
    "    elif len(alist)==0:\n",
    "        answer = \"SKIP\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('MKTIIALSYIFCLVFADYKDDDDAAAQTSAPPAVSPQITVLAPGKGPWQVAFIGITTGLLSLATVTGNLLVLISFKVNTELKTVNNYFLLSLACADLIIGTFSMNLYTTYLLMGHWALGTLACDLWLALDYVASQASVMNLLLISFDRYFSVTRPLSYRAKRTPRRAALMIGLAWLVSFVLWAPAILFWQYLVGERTVLAGQCYIQFLSQPIITFGTAMAAFYLPVTVMCTLYWRIYRETENRNIFEMLRIDEGLRLKIYKDTEGYYTIGIGHLLTKSPSLNAAKSELDKAIGRNTNGVITKDEAEKLFNQDVDAAVRGILRNAKLKPVYDSLDAVRRAALINMVFQMGETGVAGFTNSLRMLQQKRWDEAAVNLAKSRWYNQTPNRAKRVITTFRTGTWDAYFSLVKEKKAARTLSAILLAFILTWTPYNIMVLVSTFCKDCVPETLWELGYWLCYVNSTINPMCYALCNKAFRDTFRLLLLCRWDKRRWRKIPKRPGSVHRTPSRQCHHHHHH',\n",
       "  '--------------------------------------------KGPWQVAFIGITTGLLSLATVTGNLLVLISFKVNTELKTVNNYFLLSLACADLIIGTFSMNLYTTYLLMGHWALGTLACDLWLALDYVASQASVMNLLLISFDRYFSVTRPLSYRAKRTPRRAALMIGLAWLVSFVLWAPAILFWQYLVGERTVLAGQCYIQFLSQPIITFGTAMAAFYLPVTVMCTLYWRIYRETENRNIFEMLRIDEGLRLKIYKDTEGYYTIGIGHLLTKSPSLNAAKSELDKAIGRNTNGVITKDEAEKLFNQDVDAAVRGILRNAKLKPVYDSLDAVRRAALINMVFQMGETGVAGFTNSLRMLQQKRWDEAAVNLAKSRWYNQTPNRAKRVITTFRTGTWDAYFSLVKEKKAARTLSAILLAFILTWTPYNIMVLVSTFCKDCVPETLWELGYWLCYVNSTINPMCYALCNKAFRDTFRLLLLCRWDK---------------------------',\n",
       "  880.0999999999995,\n",
       "  0,\n",
       "  515)]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alignments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "alist = getListOfValidAlignments(alignments, pdb_indexes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[0]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "ialigns = []\n",
    "i =0\n",
    "alignment = alignments[i]\n",
    "alseq_fasta = alignment[0]\n",
    "alseq_pdb = alignment[1]\n",
    "\n",
    "if len(alseq_fasta)!=len(alseq_pdb):\n",
    "    print(\"Error using alignment too\")\n",
    "    print()\n",
    "    exit()\n",
    "\n",
    "last_i_pdb = -1\n",
    "last_j = -1\n",
    "i_pdb = 0\n",
    "valid = True\n",
    "for j in range(0, len(alseq_pdb)):\n",
    "        if alseq_pdb[j]!='-':\n",
    "#                                if last_i_pdb!=-1 and j - last_j > i_pdb - last_i_pdb:\n",
    "            if last_i_pdb!=-1 and j - last_j != pdb_indexes[i_pdb] - pdb_indexes[last_i_pdb]:\n",
    "                print(last_i_pdb, j, last_j, i_pdb, last_i_pdb, pdb_indexes[i_pdb], pdb_indexes[last_i_pdb])\n",
    "                valid = False\n",
    "                break\n",
    "            last_i_pdb = i_pdb\n",
    "            last_j = j\n",
    "            i_pdb = i_pdb + 1\n",
    "if valid:\n",
    "        ialigns.append(i)\n",
    "        \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
